src/datagen/main/java/mekanism/client/texture/PrideRobitTextureProvider.java

```diff
@@ -0,0 +1,143 @@
+package mekanism.client.texture;
+
+import com.google.common.hash.Hashing;
+import com.google.common.hash.HashingOutputStream;
+import mekanism.common.entity.RobitPrideSkinData;
+import net.minecraft.data.CachedOutput;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DataProvider;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.PackType;
+import net.minecraft.server.packs.resources.Resource;
+import net.minecraftforge.common.data.ExistingFileHelper;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorModel;
+import java.awt.image.WritableRaster;
```

Comment:
iirc Java AWT cannot be run on macs. While I don't use a mac we shouldn't prevent the datagenerator from being able to be ran on a mac.

Reply:
I don't think it's possible to do ImageManipulation without awt (except implementing bytereading yourself but I'm not keen on doing that)

Reply:
I will look into it some sometime in the next few days, but worst case scenario we could just use a third party library and only load it for datagen.

Reply:
Also I might be remembering wrong and awt may work on macs? Need to look into it further

Reply:
Okay I am just dumb, maybe it was awt.Color that isn't available or maybe I am thinking of some other API, but looks like MC does use awt in a variety of places so this should be relatively safe.